app/controllers/favorites_controller.rb

```diff
@@ -4,10 +4,18 @@ def index
   end
 
   def update
-    pet = Pet.find(params[:id])
-    favorite.fav_pet(pet)
+    pet = Pet.find(params[:pet_id])
```

Comment:
Better to work with IDs rather than pet objects, which get converted to hashes in the arrays